libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFAuthenticationManager.m

```diff
@@ -846,6 +846,18 @@ - (void)setupWithCredentials:(SFOAuthCredentials*) credentials {
     self.idCoordinator.delegate = self;
 }
 
+- (void)restartAuthentication {
+    @synchronized (self.authBlockList) {
+        if (!self.authenticating) {
+            [self log:SFLogLevelWarning format:@"%@: Authentication manager is not currently authenticating.  No action taken.", NSStringFromSelector(_cmd)];
+            return;
+        }
+        [self log:SFLogLevelInfo format:@"%@: Restarting in-progress authentication process.", NSStringFromSelector(_cmd)];
+        [self.coordinator stopAuthentication];
+        [self loginWithCredentials:[self createOAuthCredentials]];
```

Comment:
How do we know if the authentication in progress didn't have its own `credentials` passed in? Instead of calling `[self createOAuthCredentials]`, we should be passing in the `credentials` originally passed in here when we restart. For example, in the hybrid view controller, we explicitly pass in credentials.